modules/wallet/encrypt.go

```diff
@@ -362,6 +363,11 @@ func (w *Wallet) InitFromSeed(masterKey crypto.TwofishKey, seed modules.Seed) er
 		masterKey = crypto.TwofishKey(crypto.HashObject(seed))
 	}
 
+	if !w.scanLock.TryLock() {
+		return errScanInProgress
+	}
+	defer w.scanLock.Unlock()
+
 	w.mu.Lock()
 	defer w.mu.Unlock()
```

Comment:
yeah, didn't see much reason not to. It may not be necessary though -- seems reasonable that you should be able to check your balance and such during init. @DavidVorick, do you have any guidance on locking interactions here? Should `scanLock` always be acquired first, or after, or what?